Doesn't build against PostgreSQL 11

Hello,

I maintain an AUR package for pointcloud (https://aur.archlinux.org/packages/pgpointcloud). Today I wanted to increase the package revision to trigger a rebuild against PostgreSQL 11. I noticed that pointcloud doesn't build against pg 11.

Here is the build commands:

```
./autogen.sh
./configure --prefix=/usr
make
```

I compiled against [v1.2.0](https://github.com/pgpointcloud/pointcloud/archive/v1.2.0.tar.gz) not against master, but it doesn't seems that any commit fixed the issue on master.

Here is the full output of the build:

```
* Running /usr/bin/aclocal (1.16.1)
* Running /usr/bin/autoconf (2.69)
======================================
Now you are ready to run './configure'
======================================
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking for cpp... /usr/bin/cpp
checking for git... yes
checking for .git... no
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking how to run the C++ preprocessor... g++ -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking zlib.h usability... yes
checking zlib.h presence... yes
checking for zlib.h... yes
checking for inflate in -lz... yes
checking CUnit/CUnit.h usability... yes
checking CUnit/CUnit.h presence... yes
checking for CUnit/CUnit.h... yes
checking for CU_initialize_registry in -lcunit... yes
checking for pg_config... /usr/bin/pg_config
checking PostgreSQL version... PostgreSQL 11.1
checking libpq-fe.h usability... yes
checking libpq-fe.h presence... yes
checking for libpq-fe.h... yes
checking for PQserverVersion in -lpq... yes
checking for xml2-config... /usr/bin/xml2-config
checking libxml/tree.h usability... yes
checking libxml/tree.h presence... yes
checking for libxml/tree.h... yes
checking libxml/parser.h usability... yes
checking libxml/parser.h presence... yes
checking for libxml/parser.h... yes
checking libxml/xpath.h usability... yes
checking libxml/xpath.h presence... yes
checking for libxml/xpath.h... yes
checking libxml/xpathInternals.h usability... yes
checking libxml/xpathInternals.h presence... yes
checking for libxml/xpathInternals.h... yes
checking for xmlInitParser in -lxml2... yes
checking las.hpp usability... no
checking las.hpp presence... no
checking for las.hpp... no
configure: creating ./config.status
config.status: creating config.mk
config.status: creating pgsql/Makefile
config.status: creating lib/pc_config.h
config.status: creating pgsql/sqldefines.h

  PointCloud is now configured for 

 -------------- Compiler Info ------------- 
  C compiler:           gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt
  SQL preprocessor:     /usr/bin/cpp -traditional-cpp -w -P

 -------------- Dependencies -------------- 
  PostgreSQL config:    /usr/bin/pg_config
  PostgreSQL version:   PostgreSQL 11.1 (111)
  Libxml2 config:       /usr/bin/xml2-config
  Libxml2 version:      2.9.8
  LazPerf status:       disabled
  CUnit status:         enabled

make -C lib all
make[1]: Entering directory '/home/.../aur/pgpointcloud-aur/src/pointcloud-1.2.0/lib'
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2   -c -o pc_bytes.o pc_bytes.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2   -c -o pc_dimstats.o pc_dimstats.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2   -c -o pc_filter.o pc_filter.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2   -c -o pc_mem.o pc_mem.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2   -c -o pc_patch.o pc_patch.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2   -c -o pc_patch_dimensional.o pc_patch_dimensional.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2   -c -o pc_patch_uncompressed.o pc_patch_uncompressed.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2   -c -o pc_point.o pc_point.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2   -c -o pc_pointlist.o pc_pointlist.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2   -c -o pc_schema.o pc_schema.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2   -c -o pc_sort.o pc_sort.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2   -c -o pc_stats.o pc_stats.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2   -c -o pc_util.o pc_util.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2   -c -o pc_val.o pc_val.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2   -c -o stringbuffer.o stringbuffer.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2   -c -o hashtable.o hashtable.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2   -c -o pc_patch_lazperf.o pc_patch_lazperf.c
ar rs libpc.a pc_bytes.o pc_dimstats.o pc_filter.o pc_mem.o pc_patch.o pc_patch_dimensional.o pc_patch_uncompressed.o pc_point.o pc_pointlist.o pc_schema.o pc_sort.o pc_stats.o pc_util.o pc_val.o stringbuffer.o hashtable.o pc_patch_lazperf.o
ar: creating libpc.a
g++ -fPIC -std=c++0x -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2   -c -o lazperf_adapter.o lazperf_adapter.cpp
In file included from /usr/include/bits/libc-header-start.h:33,
                 from /usr/include/stdio.h:27,
                 from pc_api.h:16,
                 from pc_api_internal.h:18,
                 from lazperf_adapter.hpp:12,
                 from lazperf_adapter.cpp:10:
/usr/include/features.h:381:4: warning: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Wcpp]
 #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
    ^~~~~~~
ar rs liblazperf.a lazperf_adapter.o
ar: creating liblazperf.a
make -C cunit all
make[2]: Entering directory '/home/.../aur/pgpointcloud-aur/src/pointcloud-1.2.0/lib/cunit'
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2 -D_FORTIFY_SOURCE=2 -I..  -c -o cu_tester.o cu_tester.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2 -D_FORTIFY_SOURCE=2 -I..  -c -o cu_pc_bytes.o cu_pc_bytes.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2 -D_FORTIFY_SOURCE=2 -I..  -c -o cu_pc_schema.o cu_pc_schema.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2 -D_FORTIFY_SOURCE=2 -I..  -c -o cu_pc_point.o cu_pc_point.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2 -D_FORTIFY_SOURCE=2 -I..  -c -o cu_pc_patch.o cu_pc_patch.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2 -D_FORTIFY_SOURCE=2 -I..  -c -o cu_pc_patch_lazperf.o cu_pc_patch_lazperf.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2 -D_FORTIFY_SOURCE=2 -I..  -c -o cu_pc_sort.o cu_pc_sort.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -I/usr/include/libxml2 -D_FORTIFY_SOURCE=2 -D_FORTIFY_SOURCE=2 -I..  -c -o cu_pc_util.o cu_pc_util.c
gcc -o cu_tester cu_tester.o cu_pc_bytes.o cu_pc_schema.o cu_pc_point.o cu_pc_patch.o cu_pc_patch_lazperf.o cu_pc_sort.o cu_pc_util.o ../libpc.a ../liblazperf.a -lxml2 -lz -llzma -licui18n -licuuc -licudata -lm -ldl -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -lcunit -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -lz -lm -lstdc++
make[2]: Leaving directory '/home/.../aur/pgpointcloud-aur/src/pointcloud-1.2.0/lib/cunit'
make[1]: Leaving directory '/home/.../aur/pgpointcloud-aur/src/pointcloud-1.2.0/lib'
make -C pgsql all
make[1]: Entering directory '/home/.../aur/pgpointcloud-aur/src/pointcloud-1.2.0/pgsql'
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I../lib -I. -I./ -I/usr/include/postgresql/server -I/usr/include/postgresql/internal  -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2   -c -o pc_inout.o pc_inout.c
In file included from pc_pgsql.h:14,
                 from pc_inout.c:12:
pc_inout.c: In function ‘pcschema_is_valid’:
pc_inout.c:174:18: error: ‘FALSE’ undeclared (first use in this function); did you mean ‘FILE’?
   PG_RETURN_BOOL(FALSE);
                  ^~~~~
/usr/include/postgresql/server/postgres.h:387:36: note: in definition of macro ‘BoolGetDatum’
 #define BoolGetDatum(X) ((Datum) ((X) ? 1 : 0))
                                    ^
pc_inout.c:174:3: note: in expansion of macro ‘PG_RETURN_BOOL’
   PG_RETURN_BOOL(FALSE);
   ^~~~~~~~~~~~~~
pc_inout.c:174:18: note: each undeclared identifier is reported only once for each function it appears in
   PG_RETURN_BOOL(FALSE);
                  ^~~~~
/usr/include/postgresql/server/postgres.h:387:36: note: in definition of macro ‘BoolGetDatum’
 #define BoolGetDatum(X) ((Datum) ((X) ? 1 : 0))
                                    ^
pc_inout.c:174:3: note: in expansion of macro ‘PG_RETURN_BOOL’
   PG_RETURN_BOOL(FALSE);
   ^~~~~~~~~~~~~~
make[1]: *** [<builtin>: pc_inout.o] Error 1
make[1]: Leaving directory '/home/.../aur/pgpointcloud-aur/src/pointcloud-1.2.0/pgsql'
make: *** [Makefile:4: all] Error 2
```

Do you need more information to understand and fix the problem ?

--- a/pgsql/pc_access.c
+++ b/pgsql/pc_access.c
@@ -879,9 +879,9 @@ Datum pcpatch_intersects(PG_FUNCTION_ARGS)
 
 	if ( pc_bounds_intersects(&(serpa1->bounds), &(serpa2->bounds)) )
 	{
-		PG_RETURN_BOOL(TRUE);
+		PG_RETURN_BOOL(true);
 	}
-	PG_RETURN_BOOL(FALSE);
+	PG_RETURN_BOOL(false);
 }
 
 PG_FUNCTION_INFO_V1(pcpatch_size);
@@ -939,9 +939,9 @@ PG_FUNCTION_INFO_V1(pc_lazperf_enabled);
 Datum pc_lazperf_enabled(PG_FUNCTION_ARGS)
 {
 #ifdef HAVE_LAZPERF
-	PG_RETURN_BOOL(TRUE);
+	PG_RETURN_BOOL(true);
 #else
-	PG_RETURN_BOOL(FALSE);
+	PG_RETURN_BOOL(false);
 #endif
 }
 

--- a/pgsql/pc_inout.c
+++ b/pgsql/pc_inout.c
@@ -171,7 +171,7 @@ Datum pcschema_is_valid(PG_FUNCTION_ARGS)
 
 	if ( !schema )
 	{
-		PG_RETURN_BOOL(FALSE);
+		PG_RETURN_BOOL(false);
 	}
 
 	valid = pc_schema_is_valid(schema);

